python/dolfinx/wrappers/la.cpp

```diff
@@ -96,8 +96,17 @@ void la(py::module& m)
   // dolfinx::la::Vector
   py::class_<dolfinx::la::Vector<PetscScalar>,
              std::shared_ptr<dolfinx::la::Vector<PetscScalar>>>(m, "Vector")
-      .def_property_readonly("array",
-                             &dolfinx::la::Vector<PetscScalar>::mutable_array);
+      .def_property_readonly(
+          "array",
+          [](dolfinx::la::Vector<PetscScalar>& self) {
+            std::vector<PetscScalar>& array = self.mutable_array();
+            return py::array(array.size(), array.data(), py::cast(self));
+          })
+      .def_property_readonly(
```

Comment:
Why are there two near-identical functions?

Reply:
One for `dolfinx::la::Vector<PetscScalar>`  and the other for `std::shared_ptr<dolfinx::la::Vector<PetscScalar>>`.
There is possibly a smarter way to do this.. but I'm not aware of one alternative.

Reply:
I see that, but why is it needed?

Reply:
It's not actually needed.
It was just my misunderstanding of how smart pointers work in pybind.
https://pybind11.readthedocs.io/en/stable/advanced/smart_ptrs.html#std-shared-ptr

Just removed the first function.